runtime/lua/vim/lsp/util.lua

```diff
@@ -725,6 +725,7 @@ do
         [protocol.DocumentHighlightKind.Read] = "LspReferenceRead";
         [protocol.DocumentHighlightKind.Write] = "LspReferenceWrite";
       }
+      reference["kind"] = reference["kind"] or protocol.DocumentHighlightKind.Text
```

Comment:
best not to override a variable (ease debug). Use a new one like `local kind = reference["kind"] or protocol.DocumentHighlightKind.Text`